pkg/gatherers/conditional/gather_logs_of_unhealthy_pods.go

```diff
@@ -0,0 +1,108 @@
+package conditional
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/gatherers"
+	"github.com/openshift/insights-operator/pkg/gatherers/common"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// BuildGatherLogsOfUnhealthyPods collects either current or previous logs for pods firing one of the configured alerts.
+//
+// * Location in archive: conditional/unhealthy_logs/<namespace>/<pod>/<container>/[current|previous].log
+// * Since versions:
+//   * 4.10+
+func (g *Gatherer) BuildGatherLogsOfUnhealthyPods(paramsInterface interface{}) (gatherers.GatheringClosure, error) {
+	params, ok := paramsInterface.(GatherLogsOfUnhealthyPodsParams)
+	if !ok {
+		return gatherers.GatheringClosure{}, fmt.Errorf(
+			"unexpected type in paramsInterface, expected %T, got %T",
+			GatherLogsOfUnhealthyPodsParams{}, paramsInterface,
+		)
+	}
+
+	return gatherers.GatheringClosure{
+		Run: func(ctx context.Context) ([]record.Record, []error) {
+			kubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+			if err != nil {
+				return nil, []error{err}
+			}
+			records, errs := g.gatherLogsOfUnhealthyPods(ctx, kubeClient.CoreV1(), params)
+			if len(errs) > 0 {
+				return records, errs
+			}
+			return records, nil
+		},
+		CanFail: canConditionalGathererFail,
+	}, nil
+}
+
+func (g *Gatherer) gatherLogsOfUnhealthyPods(
+	ctx context.Context, coreClient v1.CoreV1Interface, params GatherLogsOfUnhealthyPodsParams,
+) ([]record.Record, []error) {
+	errs := []error{}
+	records := []record.Record{}
+
+	alertInstances, ok := g.firingAlerts[params.AlertName]
```

Comment:
Shouldn't it be considered an error?

Reply:
What exactly? That the alert couldn't be found in the map?

Reply:
I agree the condition evaluation is basically the same (check if it's in the map) and the map should be read-only once  it's created.

Reply:
well ok. Fair point....it's more a config error for me, but we can add some message to inform about this.